beacon_node/beacon_chain/src/data_availability_checker.rs

```diff
@@ -65,16 +64,20 @@ pub struct DataAvailabilityChecker<T: EthSpec, S: SlotClock> {
 ///
 /// The blobs are all gossip and kzg verified.
 /// The block has completed all verifications except the availability check.
-struct GossipAvailabilityCache<T: EthSpec> {
+struct ReceivedComponents<T: EthSpec> {
     /// We use a `BTreeMap` here to maintain the order of `BlobSidecar`s based on index.
-    verified_blobs: BTreeMap<u64, KzgVerifiedBlob<T>>,
+    verified_blobs: FixedVector<Option<KzgVerifiedBlob<T>>, T::MaxBlobsPerBlock>,
     executed_block: Option<AvailabilityPendingExecutedBlock<T>>,
 }
 
-impl<T: EthSpec> GossipAvailabilityCache<T> {
+impl<T: EthSpec> ReceivedComponents<T> {
     fn new_from_blob(blob: KzgVerifiedBlob<T>) -> Self {
-        let mut verified_blobs = BTreeMap::new();
-        verified_blobs.insert(blob.blob_index(), blob);
+        let mut verified_blobs = FixedVector::<_, _>::default();
+        // TODO: verify that we've already ensured the blob index < T::MaxBlobsPerBlock
```

Comment:
We are ensured of this from blobs that come in from gossip, but not from rpc yet afaik.
Assuming that we use this later on for caching blobs that come in from block proposals from a local EL/ builder, i think it would make sense to add the validity condition in this cache just to be safe.